x/concentrated-liquidity/position.go

```diff
@@ -84,6 +89,22 @@ func (k Keeper) GetPosition(ctx sdk.Context, poolId uint64, owner sdk.AccAddress
 	return positionStruct, nil
 }
 
+// GetUserPositions gets all the existing user positions across many pools.
+func (k Keeper) GetUserPositions(ctx sdk.Context, addr sdk.AccAddress) ([]model.Position, error) {
+	var key []byte
+	key = append(key, types.PositionPrefix...)
+	key = append(key, address.MustLengthPrefix(addr)...)
+	return osmoutils.GatherValuesFromStorePrefix(ctx.KVStore(k.storeKey), key, ParsePositionFromBz)
+}
+
+func ParsePositionFromBz(bz []byte) (position model.Position, err error) {
+	if len(bz) == 0 {
+		return model.Position{}, errors.New("position not found")
+	}
+	err = proto.Unmarshal(bz, &position)
+	return position, err
+}
```

Comment:
Isn't this the same function from the file I sent you? If so, we should just point to that instead of code duplication.

Reply:
yup it is, i didn't branch off your PR, so this function does exist right now. Still wondering if we should wait for your PR to get merged before merging this one

Reply:
Let's try merging this sooner rather than later please because this is blocking e2e work